### DISCUSSION TITLE
Percy builds pass even when percySnapshot() fails

### DISCUSSION THREAD

**Discussion by MarosPistej** [percy/cli#539] [Ideas]
# BUG
## About
We encountered issue which was even worse as it was on auto-approve branch `master`. One `cy.percySnapshot()` command failed (unknown reason, and it happen only once) but we were surprised that cypress test pass without issue, we expected that there is some assertion that cy.percySnapshot() was successful but it looks like there isn't? Result was that there was missing snapshot which was auto-approved, so in next tests we got diff that we have new snapshot even we didn't add any.

![image](https://user-images.githubusercontent.com/43110161/120996358-896b7c80-c786-11eb-9d92-fd033d8c2292.png)

**Comment by Robdel12** [comment]
Hey @MarosPistej! This is how the SDKs currently work on purpose. We're going to be adding a flag to the CLI for users to `exit 1` if any build or snapshot errors occur in the near future.

**Reply by IanVS** [reply]
Has such a flag been added?  I really would like to fail my job if the snapshots fail...

**Reply by itsjwala** [reply]
++ @IanVS @benedfit @MarosPistej 

we believe this issue is related to https://github.com/percy/percy-cypress/issues/568.

we'll be fixing ^ the issue soon in the coming weeks. Your patience’s appreciated!

**Reply by itsjwala** [reply]
Hey folks!


The modified cypress SDK can be installed from [this](https://github.com/BrowserStackCE/percy-cypress.git) git repo by using this command:
`npm i --save-dev https://github.com/BrowserStackCE/percy-cypress.git`

What it looks like in the `package.json` file:
`"@percy/cypress": "github:BrowserStackCE/percy-cypress"`

we'll be merging this in the upstream post doing some regression.

**Reply by IanVS** [reply]
Thanks @itsjwala.  Is this compatible with `npx percy snapshot`?

**Reply by itsjwala** [reply]
Hey, @IanVS I'm afraid it's not, the above workaround is specific to Percy cypress SDK. could you start a new thread and attach verbose logs so we can check?

**Reply by IanVS** [reply]
Thanks, I've created https://github.com/percy/cli/issues/1181.

**Comment by benedfit** [comment]
I would love to have this feature too, as missing snapshots on our auto-approved base branch are also causing false positives on pull-requests

**Reply by benedfit** [reply]
It would also be great if we could have a way to configure this via the UI too. Right now only changes are listed in the GitHub check as something that needs reviewing, but I'd also like to have any missing snapshots cause the need for a review too

**Comment by benedfit** [comment]
@Robdel12 any updates on https://github.com/percy/cli/discussions/539#discussioncomment-1278103
